### ISSUE TITLE
Zero dimension arrays are not working

### ISSUE THREAD

**Issue by stassats** [drmeister/clasp#63]
(aref (make-array '()))
=>
Illegal dimensions NIL for make-array

> > (make-array '(0 2))

Debugger received error of type: SIMPLE-PROGRAM-ERROR
Too many arguments error given: 4 required: 3
Error flushed.
../../src/core/lambdaListHandler.cc:58 Caught an exception while in createBindingsInScope_argArray_TPtr

**Comment by stassats** [comment]
Neither form produces errors now, but the latter is printed as `#<array 0x7fbda45c5609>`

**Comment by stassats** [comment]
And there's a problem with zero-dimensional printing and reading:
(make-array () :initial-element 10) => #0A0, should be #0A10
(aref #0A10) => NIL, should be 10

**Comment by Shinmera** [comment]
Can confirm on b05884f.
```
> (make-array '(0 2))

#<array 0x7f79fbb0d9b1>
> (make-array () :initial-element 10)

#0A0
> (aref #0A10)

NIL
```

**Comment by Shinmera** [comment]
Additionally: `(length #0A())`

**Comment by Shinmera** [comment]
Except for the `(aref #0A10)` case, everything seems fixed now.
```
> (aref (make-array '()))

NIL
> (make-array '(0 2))

#2A()
> (make-array () :initial-element 10)

#0A10
> (aref #0A10)

NIL
> (length #0A())

Condition of type: TYPE-ERROR
#0A0 is not of type SEQUENCE.
```

**Comment by Shinmera** [comment]
Confirmed fixed on dcc6f10e04e998f628a90922797eeca6dd8b68f9
